qabelbox/src/main/java/de/qabel/qabelbox/fragments/CreateIdentityMainFragment.kt

```
@@ -0,0 +1,92 @@
+        mImportIdentity = view.findViewById(R.id.bt_import_identity) as Button
+        mCreateIdentity!!.setOnClickListener(this)
+        mImportIdentity!!.setOnClickListener(this)
+        return view
+    }
+
+    override fun check(): String? {
+        return null
+    }
+
+    override fun onClick(v: View) {
+        if (v === mCreateIdentity) {
+            mActivity.handleNextClick()
+        }
+        if (v === mImportIdentity) {
+            val intent = Intent(Intent.ACTION_OPEN_DOCUMENT)
+            intent.addCategory(Intent.CATEGORY_OPENABLE)
+            intent.type = "*/*"
+            startActivityForResult(intent, CreateIdentityActivity.REQUEST_CODE_IMPORT_IDENTITY)
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int,
+                                  resultData: Intent?) {
+
+        if (requestCode == CreateIdentityActivity.REQUEST_CODE_IMPORT_IDENTITY && resultCode == Activity.RESULT_OK) {
+            importIdentity(mActivity, resultData)
+        }
+    }
+
+    fun importIdentity(activity: Activity, resultData: Intent?) {
+        resultData?.let {
+            val uri = resultData.data
+            try {
+                activity.contentResolver.openFileDescriptor(uri, "r")!!.use {
+                    identityUseCase.importIdentity(it.fileDescriptor).subscribe({
+                        toast(getString(R.string.idenity_imported))
+                        (activity as CreateIdentityActivity).apply {
+                            createdIdentity = it
+                            completeWizard()
+                        }
+                    }, {
+                        throw it
+                    })
+                }
+            } catch (e: Throwable) {
+                when (e) {
+                    is EntityExistsException ->
+                        longToast(R.string.create_identity_already_exists)
+                    else ->
+                        longToast(R.string.cant_read_identity)
```

onError is implemented as second parameter for subscribe and should throw the exception in the catch